sqlglot/parser.py

```diff
@@ -5517,16 +5518,18 @@ def _parse_id_var(
         self,
         any_token: bool = True,
         tokens: t.Optional[t.Collection[TokenType]] = None,
+        comment: bool = False,
     ) -> t.Optional[exp.Expression]:
-        identifier = self._parse_identifier()
-        if identifier:
-            return identifier
-
-        if (any_token and self._advance_any()) or self._match_set(tokens or self.ID_VAR_TOKENS):
+        expression = self._parse_identifier()
+        if not expression and (
+            (any_token and self._advance_any()) or self._match_set(tokens or self.ID_VAR_TOKENS)
+        ):
             quoted = self._prev.token_type == TokenType.STRING
-            return exp.Identifier(this=self._prev.text, quoted=quoted)
+            expression = exp.Identifier(this=self._prev.text, quoted=quoted)
+            if comment:
```

Comment:
why is this needed? why can't we just always add comments?

Reply:
self.expression(exp.Identifier, ...)

Reply:
Because the comments associated with identifiers are actually attached in columns, due to the `self.expression(exp.Column, ...)` call, which in turn calls `self._add_comments`.

Reply:
So if you prefer moving comments at the identifier level (which is a breaking change), then your suggestion would work fine. Perhaps we could route `Column.comments` to `Column.this.comments` or something.

Reply:
FWIW the behavior is now inconsistent because delimited identifiers do get comments attached to them, as I explained in https://github.com/tobymao/sqlglot/pull/3340#discussion_r1575032887. So perhaps we should either change that or make sure we also do it for non-delimited ones.